### ISSUE TITLE
Key "plugins": Key "react-refresh": Expected an object.

### ISSUE THREAD

**Issue by reyhankaplan** [ArnaudBarre/eslint-plugin-react-refresh#50]
Hello, I am getting this error when I run the lint command:

`ConfigError: Config (unnamed): Key "plugins": Key "react-refresh": Expected an object.`

I am using `eslint@9.10.0` and I have `type="module"` in my package.json, I import and use the plugin like this:

```typescript
import reactRefresh from 'eslint-plugin-react-refresh'
```

```typescript
  // ...
  {
    files: ['**/*.tsx'],
    plugins: {
      'react-refresh': reactRefresh,
    },
    rules: {
      'react-refresh/only-export-components': [
        'warn',
        { allowConstantExport: true },
      ],
    },
  },
  // ...
```

I found that it works fine if I remove the `0 &&` from the below code in the bundled index.js file or if I use `require('eslint-plugin-react-refresh')`

```javascript
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  rules
});
```

**Comment by ArnaudBarre** [comment]
What's your node version? Do have any issues when running a fresh vite react templates (also using esm + eslint v9 + this plugin)

**Comment by ArnaudBarre** [comment]
Do you have more information on your setup? tsconfig option should only impact how the TS import are resolved for type checking, but it doesn't impact node at runtime, expect if you are using something else?

**Comment by reyhankaplan** [comment]
Sorry my bad, I am using unstable [eslint.config.ts](https://eslint.org/docs/latest/use/configure/configuration-files#typescript-configuration-files)

You can check this [repo](https://github.com/reyhankaplan/eslint-config-repro) I created.

Or you can reproduce the issue using Vite with additional `eslint.config.ts` setup:
- create a new react-ts vite app `npm create vite@latest -- --template react-ts`
- rename `eslint.config.js` to `eslint.config.ts`
- install jiti `npm install -D jiti`
- run eslint `npx eslint --flag unstable_ts_config`.

--

> What's your node version?

My node version is 20.16.0

**Comment by ArnaudBarre** [comment]
This is fixed in the latest version!
